src/components/tooltip/tooltip.js

```diff
@@ -3,391 +3,428 @@
-
-            if (!scope.visibleWatcher) {
-              onVisibleChanged(scope.visible);
+        function setVisible(value) {
+          // Break if passed value is already in queue or there is no queue and
+          // passed value is current in the controller.
+          if (
+            setVisible.queued && setVisible.value === !!value ||
+            !setVisible.queued && scope.mdVisible === !!value
+          ) {
+            return;
+          }
+          setVisible.value = !!value;
+
+          if (!setVisible.queued) {
+            if (value) {
+              setVisible.queued = true;
+              showTimeout = $timeout(function() {
+                scope.mdVisible = setVisible.value;
+                setVisible.queued = false;
+                showTimeout = null;
+                if (!scope.visibleWatcher) {
+                  onVisibleChanged(scope.mdVisible);
+                }
+              }, scope.mdDelay);
+            } else {
+              $mdUtil.nextTick(function() {
+                scope.mdVisible = false;
+                if (!scope.visibleWatcher) {
+                  onVisibleChanged(false);
+                }
+              });
             }
-          }, scope.delay);
-        } else {
-          $mdUtil.nextTick(function() {
-            scope.visible = false;
-            if (!scope.visibleWatcher)
-              onVisibleChanged(false);
-          });
+          }
         }
-      }
-    }
-
-    function showTooltip() {
-      //  Do not show the tooltip if the text is empty.
-      if (!element[0].textContent.trim()) return;
-
-      // Insert the element and position at top left, so we can get the position
-      // and check if we should display it
-      element.css({top: 0, left: 0});
-      tooltipParent.append(element);
-
-      // Check if we should display it or not.
-      // This handles hide-* and show-* along with any user defined css
-      if ( $mdUtil.hasComputedStyle(element, 'display', 'none')) {
-        scope.visible = false;
-        element.detach();
-        return;
-      }
-
-      updatePosition();
 
-      $animate.addClass(content, SHOW_CLASS).then(function() {
-        element.addClass(SHOW_CLASS);
-      });
-    }
-
-    function hideTooltip() {
-      $animate.removeClass(content, SHOW_CLASS).then(function(){
-        element.removeClass(SHOW_CLASS);
-        if (!scope.visible) element.detach();
-      });
-    }
+        function onVisibleChanged(isVisible) {
+          if (isVisible) {
+            showTooltip();
+          } else {
+            hideTooltip();
+          }
+        }
 
-    function updatePosition() {
-      if ( !scope.visible ) return;
+        function showTooltip() {
+          // Do not show the tooltip if the text is empty.
+          if (!element[0].textContent.trim()) {
+            throw new Error('Text for the tooltip has not been provided. ' +
+                'Please include text within the mdTooltip element.');
```

Comment:
@DevVersion I didn't realize. What is the necessary action to take for a breaking change? Just want to make sure that I handle it properly.